### DISCUSSION TITLE
PWCoefficient on both boundary and domain

### DISCUSSION THREAD

**Discussion by lmolin3** [mfem/mfem#4520] [Q&A] [answered]
Hi All,
I've a quick question about the usage of PWCoefficients (PWMatrixCoefficient in my case).
If I understand correctly, the way keys are interpreted depends on the type of integrator the coeff is used in, whether domain or boundary.
Doesthis mean that for the same coefficient to be used both in a boundary and domain integrator I need two separate copies, one initialized with bdry attributes, the other with domain attributes?

Best,
Leo

**Comment by vladotomov** [comment, marked as answer]
Hi @lmolin3,

Yes, it looks up the attribute of the `ElementTransformation`, which can represent volumetric or boundary elements.
